cluster/charts/rook-ceph/templates/clusterrole.yaml

```diff
@@ -4,8 +4,8 @@ apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: rook-ceph-system
   labels:
-    operator: rook
-    storage-backend: ceph
+    app.kubernetes.io/operator: rook
```

Comment:
I continue to think that it isn't intended by https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/ that we use `app.kubernetes.io` as a prefix for our own descriptors.

My read of this is that Rook should apply the `app.kubernetes.io/managed-by: rook-ceph-operator` label to all resources it creates and manages, as well as the other recommended labels.

`app.kubernetes.io/created-by: rook-ceph-operator` for example
`app.kubernetes.io/component: ceph-mon` for example
`app.kubernetes.io/part-of: rook-ceph` for example
`app.kubernetes.io/version: <ceph-version>` for example
`app.kubernetes.io/name: rook-ceph-mon-a` for example

Reply:
oh, yes I take it `recommended labels` as Prefix supporter to query resources easily.

Thanks for the comment:), surely I will make the changes accordingly.

Reply:
I think perhaps there is a misunderstanding here. The original bug is not asking us to add these labels to our example/helm manifests, though I don't think it's a bad idea to do that. 

The original bug is asking to have these labels created for all the resources that the Rook operator creates, and that means modifying `Go` code. For example, when we create a monitor deployment, we need to apply something like the following labels to the deployment spec and the deployment's pod template spec:

```
app.kubernetes.io/name: ceph-mon  # name of application (I take this to mean the primary binary running)
app.kubernetes.io/instance: <cluster-namespace>_mon-a  # FULLY unique name
app.kubernetes.io/version: "<ceph version>"  # version of application running in pod
app.kubernetes.io/component: ceph-mon  # component within architecture (does not have to be unique)
app.kubernetes.io/part-of: cephclusters.ceph.rook.io_<cluster-namespace>  # a ceph cluster is identified by its namespace
app.kubernetes.io/managed-by: <operator-namespace>_rook-ceph-operator  # operator can be in any namespace, so best to be fully specific
app.kubernetes.io/created-by: <operator-namepace>_rook-ceph-operator  # operator can be in any namespace, so best to be fully specific
```

Reviewing https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels, I would suggest the above for mons.

For a harder example, let's consider RGWs

```
app.kubernetes.io/name: radosgw 
app.kubernetes.io/instance: <cluster-namespace>_rgw-<store-name>-a # what if this is too long (>63 chars)?
app.kubernetes.io/version: "<ceph version>"
app.kubernetes.io/component: ceph-rgw
app.kubernetes.io/part-of: cephobjectstores.ceph.rook.io_<cluster-namespace>_<store-name>  # an object store is identified by namespace and name, but what if this is too long (>63 chars)?
app.kubernetes.io/managed-by: <operator-namespace>_rook-ceph-operator
app.kubernetes.io/created-by: <operator-namepace>_rook-ceph-operator
```

Reply:
Updated the PR,
Thanks for explaining :)

Reply:
one doubt `app.kubernetes.io/managed-by: <operator-namespace>_rook-ceph-operator ` how to get the `<operator-namespace> `?

Reply:
I would really vote to keep these labels simple and not add the namespace info to them. There are other ways to detect which operator is managing the cluster.